securebanking-openbanking-uk-rs-simulator-server/src/main/java/com/forgerock/securebanking/openbanking/uk/rs/api/backoffice/payment/validation/services/DomesticVrpValidationService.java

```diff
@@ -0,0 +1,130 @@
+            throw new OBErrorException(OBRIErrorType.REQUEST_VRP_RISK_DOESNT_MATCH_CONSENT);
+        }
+    }
+
+    //risk - validation
+    public void validateRisk(OBRisk1 risk) throws OBErrorException {
+        if (riskValidator != null) {
+            riskValidator.validate(risk);
+        } else {
+            String errorString = "No validator!";
+            log.error(errorString);
+            throw new NullPointerException(errorString);
+        }
+    }
+
+    //if the CreditorAccount was not specified in the the consent, the CreditorAccount must be specified in the instruction
+    public void checkCreditorAccountPresentInInstructionIfNotPresentInConsent(OBCashAccountCreditor3 requestCreditorAccount, FRDomesticVrpRequest consent) throws OBErrorException {
+        if (requestCreditorAccount == null) {
+            throw new OBErrorException(OBRIErrorType.REQUEST_VRP_CREDITOR_ACCOUNT_NOT_SPECIFIED);
+        } else {
+            if (requestCreditorAccount.getIdentification() == null || requestCreditorAccount.getName() == null || requestCreditorAccount.getSchemeName() == null) {
+                throw new OBErrorException(OBRIErrorType.REQUEST_VRP_CREDITOR_ACCOUNT_NOT_SPECIFIED);
+            }
+        }
+    }
+
+    //controlParameters - validation
+    public void checkControlParameters(OBDomesticVRPInstruction instruction, FRDomesticVrpRequest consent, OBDomesticVRPControlParameters controlParameters) throws OBErrorException {
+        validateMaximumIndividualAmount(instruction, consent, controlParameters);
+    }
+
+    private void validateMaximumIndividualAmount(OBDomesticVRPInstruction instruction, FRDomesticVrpRequest consent, OBDomesticVRPControlParameters controlParameters) throws OBErrorException {
+        String instructionAmount = String.valueOf(instruction.getInstructedAmount().getAmount());
+        String instructionCurrency = String.valueOf(instruction.getInstructedAmount().getCurrency());
+        // TODO - next method must be implemented correctly depending on the control parameters rules set for the payment periodic limits and maximum
+        // individual ammount
+        validateMaximumIndividualAmount(consent, Double.valueOf(instructionAmount), instructionCurrency, controlParameters);
+    }
+
+    //MaximumIndividualAmount - validation
+    private void validateMaximumIndividualAmount(FRDomesticVrpRequest consent, Double instructionAmount, String instructionCurrency, OBDomesticVRPControlParameters controlParameters) throws OBErrorException {
+        Double consentAmount = Double.valueOf(controlParameters.getMaximumIndividualAmount().getAmount());
+        String consentCurrency = controlParameters.getMaximumIndividualAmount().getCurrency();
+        if (!(instructionAmount.compareTo(consentAmount) == 0) || !(instructionCurrency.compareTo(consentCurrency) == 0)) {
+            throw new OBErrorException(
+                    OBRIErrorType.REQUEST_VRP_CONTROL_PARAMETERS_RULES);
+        }
```

Comment:
the ASPSP must return an error with code UK.OBIE.Rules.FailsControlParameters and pass in the control parameter field that caused the error in the Field field of the error message.
Example:
```java
throw new OBErrorException(
                    OBRIErrorType.REQUEST_VRP_CONTROL_PARAMETERS_RULES,
                    VRPErrorControlParametersFields.RequestControlFields.MAX_INDIVIDUAL_AMOUNT,
                    VRPErrorControlParametersFields.ConsentControlFields.MAX_INDIVIDUAL_AMOUNT);
```

Reply:
Passed control parameter to error